### DISCUSSION TITLE
[Toggle] On/Off components

### DISCUSSION THREAD

**Discussion by erickriva** [radix-ui/primitives#2061] [Ideas]
Behavior example:

https://user-images.githubusercontent.com/2746558/229311125-f30ef7c0-1471-4b35-983b-a600d18356b5.mp4

In 'Code' part, I was thinking in something like this:
```
<Toggle.Root>
  <Toggle.On>Starred</Toggle.On>
  <Toggle.Off>Star</Toggle.Off>
</Toggle.Root>
```
or, with `asChild`:
```
<Toggle.Root asChild>
  <Button>
    <Toggle.On>Starred</Toggle.On>
    <Toggle.Off>Star</Toggle.Off>
  </Button>
</Toggle.Root>
```

There is a way to create something like this in the future?

**Comment by joaom00** [comment]
Isn't this already the [Toggle primitive](https://www.radix-ui.com/docs/primitives/components/toggle)? The API you want can be easily achieved

**Reply by erickriva** [reply]
I'm already doing this in my project, but i think it could be a feature integrated into the lib.

**Reply by sigmarsson** [reply]
@joaom00 how could I persist the toggle button states, which are situated onto a pop over control ? Each time the pop over opens the buttons states are initial, i.e. pressed == false.
